src/sif/motorscootercost.cc

```diff
@@ -68,7 +68,7 @@ constexpr float kDestinationOnlyFactor = 0.2f;
 // roads. These penalties are modulated by the road factor - further
 // avoiding higher class roads for those with low propensity for using
 // primary roads.
-constexpr float kRoadClassFactor[] = {
+float kRoadClassFactor[] = {
```

Comment:
I'm not a fan of the runtime modification of this lookup table.  A better approach that would let us keep `constexpr` here, would be to create _two_ tables.

It looks like only `use_primary` affects the result, so we could have `kRoadClassFactorWhenPrimary` and `kRoadClassFactorOtherwise`, or something of that nature.

That would avoid the per-request modification of the kRoadClassFactor lookup table.

Reply:
Thank you very much for your feedback. I am not the programmer expert and am more "digging" through the code. Sorry! 

as i mentioned above, every feedback is VERY welcome. Can you do some changes as well to the code as well?